gen/calc.py

```diff
@@ -261,7 +261,14 @@ def validate_dcos_overlay_network(dcos_overlay_network):
 
 
 def calculate_oauth_available(oauth_enabled):
-    return oauth_enabled
+    if oauth_enabled in ["[[[variables('oauthEnabled')]]]", '{ "Ref" : "OAuthEnabled" }']:
```

Comment:
These are also in this file a bit earlier, so I'd like to defer this question until later (https://github.com/dcos/dcos/blob/master/gen/calc.py#L220). This just copy/pasted that set of possible values.

I also think if `gen.calc` imported `gen.build_deploy.aws` / `gen.build_deploy.azure` there would be likely circular import issues.

The need for this also goes away entirely with #1009 since we no longer call calculate or validate functions on variables which are late.